frontend/src/pages/PokerPage.vue

```diff
@@ -1,288 +1,596 @@
+            let voteResults = this.room.stories[i].voteResults;
+            this.room.stories[i].voteResults = [];
+            for (let j = 0; j < voteResults.length; j++) {
+              if (voteResults[j].id !== this.room.currentVoteResult.id) {
+                this.room.stories[i].voteResults.push(voteResults[j]);
+              } else {
+                this.room.stories[i].voteResults.push(this.room.currentVoteResult);
+              }
+            }
+          }
+        }
+      },
+      userJoinedMessageReceived(payload) {
+        let username = this.parseWSResponseBody(payload.body);
+        let usernameObj = JSON.parse(username);
+
+        if (usernameObj["username"] !== this.username) {
+          let newArr = [];
+          for (let i = 0; i < this.room.users.length; i++) {
+            newArr.push(this.room.users[i].username);
+          }
+
+          newArr.push(usernameObj["username"]);
+          
+          newArr.sort();
+
+          this.room.users = [];
+
+          newArr.forEach((name) => this.room.users.push({username: name}));
+        }
+      },
+      userRemovedMessageReceived(payload) {
+        let username = this.parseWSResponseBody(payload.body);
+        let usernameObj = JSON.parse(username);
+        
+        let arr = [];
+
+        for (let i = 0; i < this.room.users.length; i++) {
+          if (this.room.users[i].username !== usernameObj["username"]) {
+            arr.push(this.room.users[i]);
+          }
+        }
+
+        this.room.users = arr;
+      },
+      startVoting() {
+        store.getStompClient.send("/app/poker/room/" + this.room.id + "/story/" + this.room.selectedStory["id"] + "/vote/open", {});
+      },
+      finishVoting() {
+        store.getStompClient.send("/app/poker/room/" + this.room.id + "/story/" + this.room.selectedStory["id"] + "/vote/close", {});
+      },
+      getRoom() {
+        axios.get(this.backendUrls.getRoom() + this.roomId)
+            .then(response => {
+              if (response.data != null) {
+                this.room = response.data;
+              }
+            });
+      },
+      calculateVotesAvg() {
+        if (this.room.currentVoteResult && this.room.currentVoteResult.votes) {
+          let total = 0;
+          for (let i = 0; i < this.room.currentVoteResult.votes.length; i++) {
+            total += this.room.currentVoteResult.votes[i].value;
+          }
+
+          return total / this.room.currentVoteResult.votes.length;
+        }
+        return "";
+      },
+      getVoteValue(user) {
+        if (this.room.selectedStory) {
+          if (this.room.currentVoteResult && this.room.currentVoteResult.votes) {
+            let votes = this.room.currentVoteResult.votes;
+            for (let i = 0; i < votes.length; i++) {
+              if (votes[i].username === user.username) {
+                return votes[i].value;
+              }
+            }
+          }
+        }
+        return "Not voted yet!";
```

Comment:
I think 'Voting....' and 'Not voted yet!' are not both needed, only one of them is enough.